examples/event_handling/EventHandling.cpp

```diff
@@ -0,0 +1,224 @@
+                            // m_log.emplace_back("2: Other Event");
+                        }
+                    });
+            }
+            else if (handlerType == 3)
+            {
+                // Forward to other callable
+                // In this case we forward it to our handle member functions
+                // we defined for the concrete event types we want to handle
+                // When choosing this method a default "catch-all" handler must
+                // be available for unhandled events to be forwarded to
+                // If you don't want to provide an empty "catch-all" handler
+                // you will have to make sure (e.g. via if constexpr) that this
+                // lambda doesn't attempt to call a member function that doesn't exist
+                m_window.handleEvents([this](const auto& event) { handle(event); });
+            }
+
+            // Switch to next form of event handling
+            handlerType = (handlerType + 1) % 4;
+
+            while (m_log.size() > 24u)
+                m_log.erase(m_log.begin());
+
+            // Draw the contents of the log to the window
+            m_window.clear();
+
+            for (std::size_t i = 0; i < m_log.size(); ++i)
+            {
+                m_logText.setPosition({50.f, static_cast<float>(i * 20) + 50.f});
+                m_logText.setString(m_log[i]);
+                m_window.draw(m_logText);
+            }
+
+            m_window.display();
+        }
+    }
+
+    void handle(const sf::Event::Closed&)
+    {
+        m_window.close();
+    }
+
+    void handle(const sf::Event::KeyPressed& keyPress)
+    {
+        m_log.emplace_back("3: Key Pressed: " + sf::Keyboard::getDescription(keyPress.scancode));
+    }
+
+    void handle(const sf::Event::MouseMoved& mouseMoved)
+    {
+        m_log.emplace_back("3: Mouse Moved: " + vec2ToString(mouseMoved.position));
+    }
+
+    void handle(const sf::Event::MouseButtonPressed&)
+    {
+        m_log.emplace_back("3: Mouse Pressed");
+    }
+
+    void handle(const sf::Event::TouchBegan& touchBegan)
+    {
+        m_log.emplace_back("3: Touch Began: " + vec2ToString(touchBegan.position));
+    }
+
+    template <typename T>
+    void handle(const T&)
+    {
+        // All unhandled events will end up here
+        // m_log.emplace_back("3: Other Event");
+    }
+
+private:
+    ////////////////////////////////////////////////////////////
+    // Member data
+    ////////////////////////////////////////////////////////////
+    sf::RenderWindow         m_window;
+    const sf::Font           m_font;
+    sf::Text                 m_logText;
+    std::vector<std::string> m_log;
+};
+
+
+////////////////////////////////////////////////////////////
+/// Entry point of application
+///
+/// \return Application exit code
+///
+////////////////////////////////////////////////////////////
+int main()
+{
+    Application application;
+    application.run();
+}
```

Comment:
I'd always use explicit returns in example codes.

Reply:
We stopped doing that a while ago. C++ guarantees a return value of `0` if `main` finishes so I prefer omitting that unnecessary `return 0;` line. Oddly C _does_ require it. It's a subtle difference between languages.